sites/main-site/src/content/blog/2024/seqera-containers-part-1.mdx

```diff
@@ -0,0 +1,453 @@
+</small>
+
+:::
+
+### Mulled (multi-package) images
+
+<Image
+    class="d-block ms-3 mb-3 float-end rounded w-25 shadow"
+    src={mulled}
+    alt="Photo by Hannah Pemberton on Unsplash."
+/>
+
+BioContainers is primarily set up to have a 1:1 relationship with Bioconda.
+This is great for bioinformatics tools as an image is created every time a package is published.
+This works well until you need to use more than one tool in a single process, particularly when the tools are from elsewhere in the conda ecosystem.
+For example, many bioinformatics tools leverage [samtools](https://www.htslib.org/) to convert file formats or sort reads on the fly.
+Others may pipe output to compression tools like [pigz](https://zlib.net/pigz/).
+To resolve this, BioContainers has the concept of “mulled” images (as in [mulled wine](https://en.wikipedia.org/wiki/Mulled_wine)).
+Unfortunately, generating mulled containers is not trivial.
+
+:::tip{title="How to make a mulled BioContainer" collapse}
+Luke Pembleton, Nextflow Ambassador, has a great blog post summarising the dark art of
+[Finding the right mulled biocontainer](https://lpembleton.rbind.io/posts/mulled-biocontainers/).
+Galaxy also has [documentation on the topic](https://docs.galaxyproject.org/en/master/admin/container_resolvers.html).
+
+In short, to request an image, you need to scroll to the bottom of a large
+[CSV file full of hashes](https://github.com/BioContainers/multi-package-containers/blob/master/combinations/hash.tsv)
+and add a line with your requested conda packages.
+Once edited, the images are built on CI.
+
+It used to be that you had to hunt through the GitHub action log to find the URI for your new container.
+However, [Moriz E. Beber](https://github.com/Midnighter) from the nf-core community has created a webpage to
+[generate the name of the mulled container](https://midnighter.github.io/mulled) which gently guides you through
+the process of finding the name of your container and how to update a container if you want to bump the software versions.
+
+Once the image is built and you know its address, the final step is to go to nf-core/modules,
+create a pull-request with the updated containers, and bump the versions in the conda `environment.yml`.
+:::
+
+This system works and, despite its complexity, is now a familiar process to many nf-core developers.
+However, it does present some problems.
+It's a highly manual process to update software versions and the container declarations - just fetching the software for a module often ends up taking longer than writing the entire module.
+There are also significant delays in waiting for the images to become available.
+All in all, it can be a frustrating experience and we see this in the volume of Slack messages asking for help.
+It seems likely that this puts some people off from contributing to nf-core.
+
+Migrating nf-core to use Seqera Containers will make provisioning multi-package containers trivial.
```

Comment:
```suggestion
Migrating nf-core to use Seqera Containers will hopeully make provisioning multi-package containers easier.
```

I don't like terminology such as 'trivial', as it makes assumptions on the ability of people to do stuff - for some people it might _not_ be easy, and will make them feel bad/stupid.

Reply:
Agree, but not the hopefully. It _will_ make it easier.

```suggestion
Migrating nf-core to use Seqera Containers will make provisioning multi-package containers easier.
```